misc/python/materialize/ci_util/__init__.py

```diff
@@ -49,25 +50,47 @@ def upload_junit_report(suite: str, junit_report: Path) -> None:
     ui.header(f"Uploading report for suite {suite!r} to Buildkite Test Analytics")
     suite = suite.upper().replace("-", "_")
     token = os.environ[f"BUILDKITE_TEST_ANALYTICS_API_KEY_{suite}"]
-    try:
-        res = requests.post(
-            "https://analytics-api.buildkite.com/v1/uploads",
-            headers={"Authorization": f"Token {token}"},
-            json={
-                "format": "junit",
-                "run_env": {
-                    "key": os.environ["BUILDKITE_BUILD_ID"],
-                    "CI": "buildkite",
-                    "number": os.environ["BUILDKITE_BUILD_NUMBER"],
-                    "job_id": os.environ["BUILDKITE_JOB_ID"],
-                    "branch": os.environ["BUILDKITE_BRANCH"],
-                    "commit_sha": os.environ["BUILDKITE_COMMIT"],
-                    "message": os.environ["BUILDKITE_MESSAGE"],
-                    "url": os.environ["BUILDKITE_BUILD_URL"],
-                },
-                "data": junit_report.read_text(),
+    res = requests.post(
+        "https://analytics-api.buildkite.com/v1/uploads",
+        headers={"Authorization": f"Token {token}"},
+        json={
+            "format": "junit",
+            "run_env": {
+                "key": os.environ["BUILDKITE_BUILD_ID"],
+                "CI": "buildkite",
+                "number": os.environ["BUILDKITE_BUILD_NUMBER"],
+                "job_id": os.environ["BUILDKITE_JOB_ID"],
+                "branch": os.environ["BUILDKITE_BRANCH"],
+                "commit_sha": os.environ["BUILDKITE_COMMIT"],
+                "message": os.environ["BUILDKITE_MESSAGE"],
+                "url": os.environ["BUILDKITE_BUILD_URL"],
             },
-        )
-    except Exception as e:
-        print(f"Got exception when uploading analytics: {e}")
+            "data": junit_report.read_text(),
+        },
+    )
     print(res.status_code, res.text)
+
+
+def get_artifacts() -> Any:
+    """Get artifact informations from Buildkite. Outside of CI, this function does nothing."""
+
+    if "CI" not in os.environ:
+        return []
+
+    ui.header(f"Getting artifact information from Buildkite")
+    org = os.environ["BUILDKITE_ORGANIZATION_SLUG"]
+    pipeline = os.environ["BUILDKITE_PIPELINE_SLUG"]
+    build = os.environ["BUILDKITE_BUILD_NUMBER"]
+    job = os.environ["BUILDKITE_JOB_ID"]
+    token = os.environ["BUILDKITE_AGENT_ACCESS_TOKEN"]
```

Comment:
So the token here is a [session token](https://buildkite.com/docs/agent/v3/tokens#session-tokens). As far as I can tell, the session token is only valid for use with the ["Agents REST API"](https://buildkite.com/docs/apis/agent-api), but you're trying to use the ["Pipelines REST API"](https://buildkite.com/docs/apis/rest-api/access-token) here.

I think you can just use the agent API here instead:

```py
build_id = os.environ["BUILDKITE_BUILD_ID"]
token = os.environ["BUILDKITE_AGENT_ACCESS_TOKEN"]

res = requests.get(
    f"https://agent.buildkite.com/builds/build_id/artifacts/search",
    headers={"Authorization": f"Bearer {token}"},
)
```

There's no documentation for this API, but the Go source code for the `buildkite artifact download` command, which uses this API, is quite readable: https://github.com/buildkite/agent/blob/ad143e4039beb04a9d17d11896d93cf93b2b2532/api/artifacts.go#L135-L154